listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -307,18 +347,30 @@ export default class ListenCard extends React.Component<
               </ul>
             </>
           )}
-          <button
-            title="Play"
-            className="btn-transparent play-button"
-            onClick={this.playListen}
-            type="button"
-          >
-            {isCurrentlyPlaying ? (
-              <FontAwesomeIcon size="1x" icon={faPlay as IconProp} />
-            ) : (
-              <FontAwesomeIcon size="2x" icon={faPlayCircle as IconProp} />
-            )}
-          </button>
+          {disablePlay ? null : (
+            <button
+              title="Play"
+              className="btn-transparent play-button"
+              onClick={this.playListen}
+              type="button"
+            >
+              {isCurrentlyPlaying ? (
+                <FontAwesomeIcon size="1x" icon={faPlay as IconProp} />
+              ) : (
+                <FontAwesomeIcon size="2x" icon={faPlayCircle as IconProp} />
+              )}
+            </button>
+          )}
+          {isMissingData ? (
+            <button
+              title="Add to MusicBrainz"
+              className="btn-transparent"
+              onClick={this.submitMissingData}
+              type="button"
+            >
+              <FontAwesomeIcon size="2x" icon={faPlus as IconProp} />
+            </button>
+          ) : null}
```

Comment:
Here's a proposition for another way to do this:

As you mentioned, I recently did a lot of refactoring of the ListenCard to use it everywhere, and found myself needing to be able to extend the functionality.
Instead of implementing each functionality directly in ListenCard (which would become very messy) I opted for passing some menu options as props (`additionalMenuItems`) to the ListenCard component, for any non-generic functionality.

You'll be able to move the `submitMissingData` method to the MissingMBData component and pass a menu item as prop to the ListenCard components.
You can see an example of this here: https://github.com/metabrainz/listenbrainz-server/blob/43d4c9df9cbeef480670065204859292961f70f7/listenbrainz/webserver/static/js/src/playlists/PlaylistItemCard.tsx#L64-L87

You'll probably need to bing the method like this `action={this.submitMissingData.bind(this, data)}` to ensure calling submitMissingData with the correct piece of data.
Similarly as what you did with the button, you can pass `icon={faPlus}` and the title as well.

Reply:
Ah, maybe instead of using `redirect_uri` you can set the from's `target` property to "_blank" , that should open it in a new tab. Haven't tried that myself so take with a pinch of salt :)